pkgs/development/compilers/swift/compiler/default.nix

```diff
@@ -190,6 +190,24 @@ let
     '';
   };
 
+  ninja' =
+    if stdenv.hostPlatform.isDarwin then
```

Comment:
This should probably be `buildPlatform`, right?

Reply:
Since darwin can't be cross compiled, it should function the same I believe. Though `buildPlatform` is better suited here.

Reply:
Darwin to Darwin cross-compilation was fixed in https://github.com/NixOS/nixpkgs/pull/346043.

Reply:
The real question is whether it breaks compiling a Linux Swift on Darwin. (Probably that doesn’t quite work right now for unrelated reasons, is my guess.)

Anyway if we pin we should just pin unconditionally to match upstream.